sql/mysqld.cc

```diff
@@ -7691,6 +7707,13 @@ static int option_cmp(my_option *a, my_option *b)
         return 1;
     }
   }
+
+  if (deprecated_a)
+    return -1;
+
+  if (deprecated_b)
+    return 1;
+
```

Comment:
The idea behind this is to print the deprecated variables grouped with the corresponding deprecation substitute, so if deprecation substitute is set we'll use that within the sort function (instead of the variable name) to group them together.